### PR TITLE
add office macro execution vst

### PR DESCRIPTION
**Pull request by VVX7**

### CODE REVIEW

**Comment by VVX7** [comment]
@privateducky this check the HKCU so it only works for the user the probe is running as.  we could load other user hives but the probe would need to run as root in that case.

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	"golang.org/x/sys/windows/registry"
+)
+
+func test() {
+	exit := 101
+	applications := [3]string{"Word", "Excel", "PowerPoint"}
+
+	for _, app := range applications {
+		fmt.Printf("[*] Checking %s macro-enabled documents...\n", app)
```
Comment (privateducky) [inline tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:20]: use println to avoid the import

Comment (privateducky) [reply tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:20]: and the \n

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/*
+NAME: 24f142c5-9fa5-4da3-bd58-8294bb7a1742.go
+RULE: A macro-enabled document was executed
```
Comment (privateducky) [inline tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:3]: What is the goal of this test?

**Review by privateducky — CHANGES_REQUESTED** [review]
Comments

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go (hunk 2 above)
Comment (VVX7) [reply tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:3]: The test is successful if a user has clicked 'enabled content' on an Office macro document. The goal is know whether such a document has been executed on this by the current user.

Comment (privateducky) [reply tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go]: oooh I see; this is perfect. 
very curious how testing on this goes.

**Comment by bfuzzy1** [comment]
@VVX7 Pinged you on error handling already.

**Comment by VVX7** [comment]
@bfuzzy1 Yep! I'll change this to handle errors so it's consistent with Detect exit codes. `registry.OpenKey` just bubbles up a Windows error code. the test should should check if the error means the key doesnt exist and carry on, or panic otherwise.  just making sure we're good with using the `golang.org/x/sys/windows/registry` import first.

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go (hunk 2 above)
Comment (privateducky) [reply tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go]: test on all platforms ;)

Comment (bfuzzy1) [reply tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go]: oh yea 100% I just tested that on Windows 11 real quick lol

**Comment by VVX7** [comment]
added better error handling. will now `104` on non-windows OS and throw an `1` if a reg key value isn't readable and a macro isn't found.  if a macro is found and some reg keys are not readable the test still returns `100`.

@privateducky how do you feel about `regexp` import (used to match the Office version) and `runtime` which we already use in pneuma to test the platform the agent is running on?

**Review by privateducky — CHANGES_REQUESTED** [review]
@VVX7 what's the status of this test? it's a great test and would love to get it in.. but it needs to be updated with:
- the start/stop functions in endpoint
- compile flags to ensure this compiles only for windows
- tested

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
+/*
+NAME: macro.go
```
Comment (privateducky) [inline tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:2]: swap NAME for ID

**Comment by VVX7** [comment]
this is tested and confirmed working on windows 11 arm64.

added endpoint.start, endpoint.stop, windows-only compile flag.

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
+//go:build windows || !darwin || !linux
+// +build windows !darwin !linux
+
+/*
+NAME: 24f142c5-9fa5-4da3-bd58-8294bb7a1742.go
```
Comment (privateducky) [inline tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:5]: swap name for ID

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+RULE: Has a macro-enabled document been executed
```
Comment (privateducky) [inline tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:6]: select from one of our 'top 10' rules, or see me about adding a new one

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	"golang.org/x/sys/windows/registry"
+	"os"
+	"regexp"
+	"runtime"
+
+	"github.com/preludeorg/test/endpoint"
+)
+
+func test() {
+	if runtime.GOOS != "windows" {
```
Comment (privateducky) [inline tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:22]: the compile flags at the top should limit this to only windows VST compilations, correct? if yes, we can drop this if/block.

File: tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+	exit := 101
+	errors := 0
+	applications := [3]string{"Word", "Excel", "PowerPoint"}
+	office_path := "Software\\Microsoft\\Office"
+
+	k, err := registry.OpenKey(registry.CURRENT_USER, office_path, registry.QUERY_VALUE|registry.ENUMERATE_SUB_KEYS)
+	if err != nil {
+		if err.Error() == "The system cannot find the file specified." {
+			println("Office is not installed.")
```
Comment (privateducky) [inline tests/24f142c5-9fa5-4da3-bd58-8294bb7a1742/24f142c5-9fa5-4da3-bd58-8294bb7a1742.go:35]: if no office installed, should this be exited as 104?

**Comment by privateducky** [comment]
closing, as this test is speculative

**Pull request closed without merging**